cocos/renderer/CCTextureCache.cpp

```diff
@@ -141,31 +168,30 @@ void TextureCache::addImageAsync(const std::string &path, const std::function<vo
         return;
     }
 
-    // lazy init
-    if (_loadingThread == nullptr)
-    {
-        // create a new thread to load images
-        _loadingThread = new (std::nothrow) std::thread(&TextureCache::loadImage, this);
-        _needQuit = false;
-    }
+    // generate async struct
+    AsyncStruct *data = new (std::nothrow) AsyncStruct(fullpath, callback);
 
-    if (0 == _asyncRefCount)
+    enqueueAsyncStruct(data);
+}
+
+void TextureCache::initImageAsync(Texture2D* texture, const std::string &path, const std::function<void(Texture2D*)>& callback)
+{
+    CCASSERT(texture != nullptr, "TextureCache: texture MUST not be nil");
+
+    std::string fullpath = FileUtils::getInstance()->fullPathForFilename(path);
+
+    // check if file exists
+    if ( fullpath.empty() || ! FileUtils::getInstance()->isFileExist( fullpath ) )
     {
-        Director::getInstance()->getScheduler()->schedule(CC_SCHEDULE_SELECTOR(TextureCache::addImageAsyncCallBack), this, 0, false);
+        if (callback) callback(nullptr);
+        return;
     }
 
-    ++_asyncRefCount;
-
     // generate async struct
     AsyncStruct *data = new (std::nothrow) AsyncStruct(fullpath, callback);
+    data->texture = texture;
```

Comment:
very much the same as TextureCache::addImageAsync, but will not change TextureCache._textures